test/perf/benchmarks/Program.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using BenchmarkDotNet.Running;
+using System.IO;
+using BenchmarkDotNet.Extensions;
+
+namespace MicroBenchmarks
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            var argsList = new List<string>(args);
+            int? partitionCount;
+            int? partitionIndex;
+            List<string> exclusionFilterValue;
+            List<string> categoryExclusionFilterValue;
+            bool getDiffableDisasm;
+
+            // Parse and remove any additional parameters that we need that aren't part of BDN (BenchmarkDotnet)
+            try {
+                CommandLineOptions.ParseAndRemoveIntParameter(argsList, "--partition-count", out partitionCount);
+                CommandLineOptions.ParseAndRemoveIntParameter(argsList, "--partition-index", out partitionIndex);
+                CommandLineOptions.ParseAndRemoveStringsParameter(argsList, "--exclusion-filter", out exclusionFilterValue);
+                CommandLineOptions.ParseAndRemoveStringsParameter(argsList, "--category-exclusion-filter", out categoryExclusionFilterValue);
+                CommandLineOptions.ParseAndRemoveBooleanParameter(argsList, "--disasm-diff", out getDiffableDisasm);
+
+                CommandLineOptions.ValidatePartitionParameters(partitionCount, partitionIndex);
```

Comment:
to be honest with you I don't think that you are going to need any of these custom arguments in the near future:

* we have added `--partition-count` and `--partition-index` after our portfolio has reached three thousands of benchmarks and running all of them was taking 5 hours. The partitioning allows for distributing this work amongst many machines.
* we have added `--exclusion-filter` and `--category-exclusion-filter` after WebAssembly (WASM) supports was added to .NET Runtime and it turned out that we could not run multithreaded benchmarks for WASM because it's single threaded by design.
* `--disasm-diff` this is something that we have added just to simplify the process of exporting generated disassembly to our ReportingSystem. If you would also like to do that we should just add it to BenchmarkDotNet and remove the reflection hacks.

Reply:
That's good to know. As for the `--disasm-diff` flag, we'd like to generate the same reporting as .NET performance, so that sounds like something we may need.

Reply:
The `--partition-count` and `--partition-index` sound useful to keep, though we won't need it for the foreseeable future. I hope we can also grow the number of our perf tests to the point that we need to use those 2 flags 😄

The `--exclusion-filter` and `--category-exclusion-filter` are also useful to us. We are planning to have `Interal` and `Public` categories, for benchmarks targeting the internal APIs and public APIs respectively. For benchmarks that targets the internal APIs, they won't run with the existing `7.0.x` and `7.1.x` PowerShell SDK packages because it requires `InternalVisible` attribute to be declared in the PowerShell assemblies. So we may need those 2 flags when running our benchmarks targeting the `7.0.x` and `7.1.x` packages.

Reply:
> where is the reflection hack?

https://github.com/dotnet/performance/blob/ba63f00d5a6ace5fcabb4d9ba20eecdee3925b52/src/harness/BenchmarkDotNet.Extensions/DiffableDisassemblyExporter.cs#L62-L88